include/uapi/sound/sof/header.h

```diff
@@ -26,6 +26,30 @@ struct sof_abi_hdr {
 	__u32 data[];		/**< Component data - opaque to core */
 }  __packed;
 
+/*
+ * Identifies data type, size and ABI for IPC4.
+ * Used by any bespoke component data structures or binary blobs.
+ */
+struct sof_ipc4_abi_hdr {
+	__u32 magic;		/**< 'S', 'O', 'F', '4' */
+	__u32 rsvd1;
+	__u32 size;		/**< size in bytes of data excl. this struct */
+	__u32 abi;		/**< SOF ABI version */
+	__u32 ipc_blob_type;	/** IPC4 type. 0: init, 2: config, 4: large_config */
```

Comment:
you are missing a < if you want to be consistent with your comments.

what does IPC4 type mean? 

I think this refers to when this blob needs to be used, it's not a 'type' proper.

Reply:
I would suggest you use enum or defines for the `ipc_blob_type`, which is much clearer. the test on
if (header->ipc_blob_type == SOF_IPC4_BLOB_FOR_INIT) is better than  if (header->ipc_blob_type == 1)

Reply:
Regarding the comment format, we usually use `/* ... */` in kernel, `/**< ... */` looks like was copied from firmware.

Reply:
@aiChaoSONG ```ipc_blob_type``` is using the same value of IPC4 command. We won't create a new "SOF_IPC4_BLOB_FOR_xxx" or "0". Please refer https://github.com/thesofproject/linux/pull/3755/commits/9f33ad9217358a2dced2ea0789fe7001b1687c7c#diff-694481cfb73dd1682c248c54855215595dc96ca4ee65b1d3ef9cdc4a478158e0R844 for the usage